pkg/audit/disqualification_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/pkg/audit"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+)
+
+// TestDisqualificationTooManyFailedAudits does the following:
+// * Create a failed audit report for a storagenode
+// * Record the audit report several times and check that the node isn't
+//	 disqualified until the audit reputation reaches the cut-off value.
+func TestDisqualificationTooManyFailedAudits(t *testing.T) {
+	var auditDQCutOff float64
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, Reconfigure: testplanet.Reconfigure{
+			Satellite: func(log *zap.Logger, index int, config *satellite.Config) {
+				auditDQCutOff = config.Overlay.Node.AuditReputationDQ
+				// TODO: if/v3-1952-h1 this setting will go away once
+				// https://github.com/storj/storj/pull/2271 be merge
+				config.Audit.MaxRetriesStatDB = 1
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		err := planet.Satellites[0].Audit.Service.Close()
+		require.NoError(t, err)
+
+		var (
+			sat    = planet.Satellites[0]
+			nodeID = planet.StorageNodes[0].ID()
+			report = &audit.Report{
+				Fails: storj.NodeIDList{nodeID},
+			}
+		)
+
+		dossier, err := sat.Overlay.Service.Get(ctx, nodeID)
+		require.NoError(t, err)
+
+		prevReputation := calculateStorageNodeReputation(dossier)
+
+		// Report the audit failure until the node gets disqualified due to many
+		// failed audits
+		for n := 0; ; n++ {
+			_, err := sat.Audit.Service.Reporter.RecordAudits(ctx, report)
+			require.NoError(t, err)
+
+			dossier, err := sat.Overlay.Service.Get(ctx, nodeID)
```

Comment:
This can also go inside the `calculateStorageNodeReputation` (how about just `calcReputation` as function name) function, which can take `planet` and `nodeID` as arguments, and return reputation and error.

Reply:
I prefer `calcReputation` too, however I did the long name because I'm not understanding when to use abbreviations and when not and when to use more explicit and less explicit names, etc.

I'll do those changes.

Reply:
Well, no!

I need the dossier in the loop so doing so it will require to get the dossier  twice in the loop.

Reply:
Oh, yes. You are right.

Reply:
Regarding the names, I think the rule of thumb is that we use longer names if a shorter one would be ambiguous.

Reply:
yeah! but ambiguity and abbreviations (except fa few very specific) are quite subjective so it's difficult to see when yes and no unless that you see some that they are applied.